### PR TITLE
Lint Section: A through D

### PR DESCRIPTION
**Pull request by ChaoticBoredom**
Changes to the Lint section, affecting the following cops:

- Lint/AmbiguousBlockAssociation
- Lint/AmbiguousOperator
- Lint/AmbiguousRegexpLiteral
- Lint/AssignmentInCondition
- Lint/BigDecimalNew
- Lint/BooleanSymbol
- Lint/CircularArgumentReference
- Lint/Debugger
- Lint/DeprecatedClassMethods
- Lint/DuplicateCaseCondition
- Lint/DuplicateMethods
- Lint/DuplicatedKey

Strategy was iterate through the list, checking for entries throughout the file. If entry was found, move it to roughly right place. If not, should it be enabled? If unclear, tag for discussion.

As an added bonus, I put all bad examples into a scratch file in this folder, and checked to see if rubocop captured them accurately. I also ran all of these against Jobber, and the only ones that passed were `Lint/CircularArgumentReference`, `Lint/Lint/DuplicateCaseCondition` and `Lint/Lint/DuplicatedKey` 😢 

For these, the ones worth documenting are not currently enabled IMO, but documentation should be added if they are enabled after discussion.

PTAL
@JeffMarvin @C-Flatla 
NoteI'm waiting on #12 to be merged first.

### CODE REVIEW

File: .rubocop.yml [hunk 1]
```diff
@@ -763,22 +757,21 @@ Metrics/PerceivedComplexity:
 ### Warnings
 
 Lint/AmbiguousOperator:
-  Description: >-
-                 Checks for ambiguous operators in the first argument of a
-                 method invocation without parentheses.
   Enabled: false
 
+# TODO: 2018/08/20 Discuss at next style guide meeting
+# Pretty sure we want to enable?
```
Comment (ChaoticBoredom) [inline .rubocop.yml]: This is what I'm doing to remind myself to raise at the SepMoCoReStyGuCo.
Thoughts?

File: .rubocop.yml [hunk 2]
```diff
@@ -763,22 +757,21 @@ Metrics/PerceivedComplexity:
...
   Enabled: false
 
+# TODO: 2018/08/20 Discuss at next style guide meeting
+# VERY sure we want to enable this rule
 Lint/AssignmentInCondition:
-  Description: "Don't use assignment in conditions."
   Enabled: false
   AllowSafeAssignment: true
 
+################ Things below this are not reviewed/corrected ########
```
Comment (ChaoticBoredom) [inline .rubocop.yml]: Just a note to try an keep things sane.

File: .rubocop.yml [hunk 3]
```diff
@@ -763,22 +757,21 @@ Metrics/PerceivedComplexity:
 ### Warnings
```
Comment (ChaoticBoredom) [inline .rubocop.yml:757]: Not sure if we care about keeping the headers above this.

**Comment by jessevanherk** [comment]
we've already got drift between this and our jobber repo, and I believe there's better ways to proceed on this. closing this off for now.

**Pull request closed without merging**